### DISCUSSION TITLE
Storage-Initializer initContainer not compatible with ISTIO sidecars

### DISCUSSION THREAD

**Discussion by abdulvirta** [kubeflow/kfserving#2250] [Q&A]
Hi all, I was stuck with this error where storage-initializer is not able to communicate with gs:// paths to copy the model locally. After lots of search, I found this ISTIO issue with init containers. https://istio.io/latest/docs/setup/additional-setup/cni/#compatibility-with-application-init-containers

The suggested workaround above is to have the init containers runas 1337 to avoid proxy traffic. However, we don't have control over this init container from the InferenceService CRD.

Was wondering if anyone has a workaround and what usually is done in this case? any help or hint would be greatly appreciated!

**Comment by elukey** [comment]
Hi! IIRC I faced a similar issue with the storage initializer not being able to contact Swift/S3 (on premise endpoints) and I discovered that it was due to the storage initializer running before the istio sidecar (hence before it was able to set up iptables settings to redirect traffic etc..). I used `excludeIPRanges` in my config.yaml for istioctl, under global -> proxy, to exclude some IP ranges from the istio sidecar settings.

**Comment by abdulvirta** [comment]
sorry just saw this. Yes I also got it working by adding excludeIPRanges but to the annotation of inference-service yaml. And then it passed it a long to the pods. Worked for me. Thanks!
